extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetTools.java

```diff
@@ -954,16 +958,36 @@ public static Table readPartitionedTableWithMetadata(
     public static Table readKeyValuePartitionedTable(
             @NotNull final File directory,
             @NotNull final ParquetInstructions readInstructions) {
-        return readPartitionedTable(
-                new ParquetKeyValuePartitionedLayout(directory, MAX_PARTITIONING_LEVELS_INFERENCE, readInstructions),
-                readInstructions);
+        return readPartitionedTable(new ParquetKeyValuePartitionedLayout(directory, MAX_PARTITIONING_LEVELS_INFERENCE,
+                readInstructions), readInstructions);
+    }
+
+    /**
+     * Creates a partitioned table via the key-value partitioned parquet files from the root {@code directory},
+     * inferring the table definition from those files.
+     *
+     * <p>
+     * Callers wishing to be more explicit and skip the inference step may prefer to call
+     * {@link #readKeyValuePartitionedTable(String, ParquetInstructions, TableDefinition)}.
+     *
+     * @param directory the path or URI for the root directory to search for .parquet files
+     * @param readInstructions the instructions for customizations while reading
+     * @return the table
+     * @see ParquetKeyValuePartitionedLayout#ParquetKeyValuePartitionedLayout(URI, int, ParquetInstructions)
+     * @see #readPartitionedTable(TableLocationKeyFinder, ParquetInstructions)
+     */
+    public static Table readKeyValuePartitionedTable(
```

Comment:
This is one of the crappy parts of Java IMO, the sprawl of overload, or overload-like methods; this is 1-of-4 new methods.

I don't know if we want to consider a builder-based approach to cut down on the number of permutations. More of a question / comment for @rcaudy.

Reply:
I've been increasingly frustrated with the overflowing number of overloads in `ParquetTools`. I'm not at all opposed to introducing parameter structures created via builders, just wary of API-breaking changes.

Reply:
This can be a major breaking refactoring across all methods, so I am slightly worried of making the change here.
I can discuss this on call with Ryan during the code review.